### PR TITLE
110 review and test voice cloning task

### LINKED ISSUE #110: Review and test voice cloning task
### Description

This issue is to track feedback for the senselab voice cloning task.

### Tasks

- [x]  Review tutorial
- [x]  Review documentation
- [x]  Review code
- [x]  Test edge cases

### Freeform Notes

_No response_

### PR DESCRIPTION
**Pull request by fabiocat93**
PR to address the questions in https://github.com/sensein/senselab/issues/110

### CODE REVIEW

**Comment by fabiocat93** [comment]
hi @900miles, can you please review this? 
- I have added some proper doc introducing the task and fixed the docstring and made it in g-style
- adjusted the problems with `**kwargs`
- made the knnvc error messages more specific
- added tests with more than 2 elements

File: src/senselab/audio/tasks/voice_cloning/api.py [hunk 1]
```diff
@@ -17,27 +15,39 @@ def clone_voices(
     device: Optional[DeviceType] = None,
     **kwargs: Any,  # noqa:ANN401
 ) -> List[Audio]:
-    """Clones voices from source audios to target audios using the given model."""
+    """Clones voices from source audios to target audios using the given model.
```
Comment (900miles) [inline src/senselab/audio/tasks/voice_cloning/api.py:18]: I think we can be a little clearer on exactly what is happening in the function. That this does a pairwise conversion from each audio in the source list onto the corresponding audio in the target list.

File: src/senselab/audio/tasks/voice_cloning/api.py [hunk 2]
```diff
@@ -17,27 +15,39 @@ def clone_voices(
...
+
+    Args:
+        source_audios (List[Audio]): A list of source audio samples.
+        target_audios (List[Audio]): A list of target audio samples.
+        model (SenselabModel, optional): The model to use for voice cloning.
+            Defaults to TorchModel(path_or_uri="bshall/knn-vc", revision="master").
```
Comment (900miles) [inline src/senselab/audio/tasks/voice_cloning/api.py]: We should specify in the documentation itself that as of now, only KNNVC is supported.

**Review by 900miles — COMMENTED** [review]
Looks great! Just a couple comments for clarity.

**Comment by fabiocat93** [comment]
@900miles thank you. i think i have fixed the last comments. feel free to approve the code if you think it's ready to be merged.

File: tutorials/voice_cloning.ipynb
Comment (900miles) [inline tutorials/voice_cloning.ipynb:1]: The colab link in this file points to the speech-to-text tutorial

File: src/senselab/audio/tasks/voice_cloning/api.py [hunk 3]
```diff
@@ -17,27 +15,45 @@ def clone_voices(
     device: Optional[DeviceType] = None,
     **kwargs: Any,  # noqa:ANN401
 ) -> List[Audio]:
-    """Clones voices from source audios to target audios using the given model."""
+    """Clones voices from source audios to target audios using the given model.
+
+    This function performs a pairwise voice cloning operation, where each audio in the
+    `source_audios` list is converted and mapped onto the corresponding audio in the
+    `target_audios` list. The resulting list contains target audio samples with their
+    voices replaced by the voices from the corresponding source samples.
```
Comment (900miles) [inline src/senselab/audio/tasks/voice_cloning/api.py]: I'm pretty sure this is backwards. The resulting list contains the content of the source audio file with the voice changed to sound like the voice in the corresponding target audio file, right?

**Review by 900miles — CHANGES_REQUESTED** [review]
Thanks, just a couple more comments

File: src/senselab/audio/tasks/voice_cloning/api.py (hunk 3 above)
Comment (fabiocat93) [reply src/senselab/audio/tasks/voice_cloning/api.py]: nice catch

**Comment by fabiocat93** [comment]
> Thanks, just a couple more comments

It should be ready for another review @900miles

**Review by 900miles — APPROVED** [review]
Looks great, thanks @fabiocat93!
